src/functions/__tests__/formattedEnum.test.ts

```diff
@@ -0,0 +1,137 @@
+import { AssertionError } from 'assert';
+import { formattedEnum } from '../formattedEnum';
+
+const defaultReportingThreshold = 3;
+
+function runFormattedEnum(targetVal: any, reportingThreshold: any) {
+  return formattedEnum(
+    targetVal,
+    { reportingThreshold },
+    { given: ['$'] },
+    { given: null, original: null, resolved: {} as any },
+  );
+}
+
+describe('formattedEnum', () => {
+  describe('parameters validation', () => {
+    test.each([1, { a: 1 }, 'nope', undefined])(
+      'should return undefined when the enum is not an array (%s)',
+      enumContent => {
+        const schema = {
+          type: 'integer',
+          enum: enumContent,
+          format: 'int64',
+        };
+
+        expect(runFormattedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+      },
+    );
+
+    test.each([0, -1, { a: 1 }, 'nope', undefined])(
+      'should throw when "reportingThreshold" bears an invalid value (%s)',
+      threshold => {
+        const schema = {
+          type: 'integer',
+          enum: [1, 2, 3],
+          format: 'int64',
+        };
+
+        expect(() => runFormattedEnum(schema, threshold)).toThrowError(AssertionError);
+      },
+    );
+  });
+
+  test('should return undefined when the enum contains no value', () => {
+    const schema = {
+      type: 'integer',
+      enum: [],
+      format: 'int64',
+    };
+
+    expect(runFormattedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+  });
+
+  describe('basic', () => {
+    test('should return undefined when all enum values respect the format', () => {
+      const schema = {
+        type: 'integer',
+        enum: [123, 456],
+        format: 'int64',
+      };
+
+      expect(runFormattedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+    });
+
+    test('should identify enum values that do not respect the format', () => {
+      const schema = {
+        type: 'integer',
+        enum: [123, 'a string!', 456, 'and another one!'],
+        format: 'int64',
+      };
+
+      expect(runFormattedEnum(schema, defaultReportingThreshold)).toEqual([
+        {
+          message: 'Some enum values do not respect the specified format "int64": "a string!", "and another one!".',
+        },
+      ]);
+    });
+  });
+
+  describe('formats', () => {
+    const testCases: Array<[string, string, unknown[], unknown]> = [
```

Comment:
Note to self: Add more test coverage to cover the following fomats: `int32`, `int64`, `float`, `double`, `byte`, `date`, `time`, `date-time`, `uri`, `uri-reference`, `uri-template`, `email`, `hostname`, `ipv4`, `ipv6`, `regex`, `uuid`, `json-pointer`, `relative-json-pointer`